homeassistant/components/automation/config.py

```diff
@@ -69,24 +74,54 @@ async def _try_async_validate_config_item(hass, config, full_config=None):
     return config
 
 
+def _deprecated_legacy_mode(config):
+    legacy_names = []
+    legacy_unnamed_found = False
+
+    for cfg in config[DOMAIN]:
+        mode = cfg.get(CONF_MODE)
+        if mode is None:
+            cfg[CONF_MODE] = SCRIPT_MODE_LEGACY
+            name = cfg.get(CONF_ID) or cfg.get(CONF_ALIAS)
+            if name:
+                legacy_names.append(name)
+            else:
+                legacy_unnamed_found = True
+
+    if legacy_names or legacy_unnamed_found:
+        msgs = []
+        if legacy_unnamed_found:
+            msgs.append("unnamed automations")
+        if legacy_names:
+            if len(legacy_names) == 1:
+                base_msg = "this automation"
+            else:
+                base_msg = "these automations"
+            msgs.append(f"{base_msg}: {', '.join(legacy_names)}")
+        warn_deprecated_legacy(_LOGGER, " and ".join(msgs))
```

Comment:
Tagged as a breaking change because people need to make changes.

btw, we should make sure that we update the automation and script editors. We should create an issue in the frontend for that.

CC @bramkragten